lib/Serialization/Deserialization.cpp

```diff
@@ -5328,9 +5344,22 @@ ModuleFile::loadAllConformances(const Decl *D, uint64_t contextData,
   DeclTypeCursor.JumpToBit(bitPosition);
 
   while (numConformances--) {
-    auto conf = readConformance(DeclTypeCursor);
-    if (conf.isConcrete())
-      conformances.push_back(conf.getConcrete());
+    auto conformance = readConformanceChecked(DeclTypeCursor);
+
+    if (!conformance) {
+      // Missing module errors are most likely caused by an
+      // implementation-only import hiding types and decls.
+      // rdar://problem/53116484
+      if (conformance.errorIsA<XRefNonLoadedModuleError>()) {
+        consumeError(conformance.takeError());
```

Comment:
I'm worried that this one is *not* sound. Is it okay to drop arbitrary conformances from a Decl when types may already be formed depending on those conformances? Or is that okay because anything that would have depended on the conformance would itself not be part of the ABI? @slavapestov

Reply:
I'm not really sure what it means for something to depend on the conformance in terms of ABI. Can you elaborate further?

Reply:
Can you form a `Set<Foo>` if you can't deserialize `Foo: Hashable`? And if you don't form the `Set<Foo>` and it's a stored property in a frozen struct, what happens?

Reply:
Conversely, if you already *have* `Set<Foo>` and then find out that you can't deserialize `Foo: Hashable`, what happens when you pass `Set<Foo>` to a generic method?